### ISSUE TITLE
Rename __filterChanged to filterChanged

### ISSUE THREAD

**Issue by dotpointer** [Neovici/cosmoz-omnitable#336]
By design `__filterChanged` needs to be overridden, but it needs to be renamed to `filterChanged` because it is private.

**Comment by cristinecula** [comment]
Can you give me an example where you would override `__filterChanged`?

**Comment by dotpointer** [comment]
> Can you give me an example where you would override `__filterChanged`?

This issue was requested by @nomego so he knows more, but the problem appears in `cosmoz-omnitable-column-boolean.js` when running `yarn lint`, then it prints:

`cosmoz-omnitable-column.js(84,2) warning [overriding-private] - Overriding private member '__filterChanged' inherited from columnMixin`

**Comment by cristinecula** [comment]
@nomego Looking at the code, it becomes apparent that the purpose of columnMixin's `__filterChanged` is to notify the `cosmoz-column-filter-changed` event. Looking at the usage in `cosmoz-omnitable-column`, the method is overridden to add a behavior unrelated to the original purpose:

https://github.com/Neovici/cosmoz-omnitable/blob/master/cosmoz-omnitable-column.js#L84-L89

I'm not sure if making the private method public is the best solution. Maybe it would be better to keep it private and instead set up a different observer in cosmoz-omnitable-column, just like in cosmoz-omnitable-column-boolean:

https://github.com/Neovici/cosmoz-omnitable/blob/master/cosmoz-omnitable-column-boolean.js#L90-L118

**Comment by nomego** [comment]
So now Boolean is broken cause it doesn't fire the event? (Haven't looked at the code)

I haven't actually analyzed the logic at all, just gave the answer to solve the warning.

If the nature of the observer is internal and not meant to be overridden we should change Boolean to not override it.

**Comment by cristinecula** [comment]
@nomego no, boolean has both `__filterChanged` (inherited from mixin) and `_filterChanged` (own method, notice the single underscore).

**Comment by megheaiulian** [comment]
Not applicable. Codebase has changed a lot since.
